src/idl_gen_cpp.cpp

```diff
@@ -233,6 +233,28 @@ class CppGenerator : public BaseGenerator {
     return keywords_.find(name) == keywords_.end() ? name : name + "_";
   }
 
+  std::string Name(const FieldDef &field) const {
+    // the union type field suffix is immutable
+    static size_t union_suffix_len = strlen(UnionTypeFieldSuffix());
+    // early return if no case transformation required
+    if (opts_.cpp_object_api_field_case != IDLOptions::Case_Upper &&
+	opts_.cpp_object_api_field_case != IDLOptions::Case_Lower)
+      return EscapeKeyword(field.name);
+    std::string name = field.name;
+    if (field.value.type.base_type == BASE_TYPE_UTYPE &&
+	name.length() > union_suffix_len)
+      // do not change the case style of the union type field suffix
+      name.erase(name.length() - union_suffix_len, union_suffix_len);
+    if (opts_.cpp_object_api_field_case == IDLOptions::Case_Upper)
```

Comment:
I think all the single-line if statements require `{}`, did you run the clang formatter on it?

Reply:
yes, it's fine, these are single statements

Reply:
... and travis CI rejects any clang formatter violations

Reply:
I see, i guess we set `AllowShortBlocksOnASingleLine=true` in our clang-format.